bcipy/helpers/convert.py

```diff
@@ -167,23 +285,103 @@ def write_edf(output_path: str,
             channel_info.append(ch_dict)
             data_list.append(raw_data[i])
 
+        if events:
+            writer.set_number_of_annotation_signals(annotation_channels)
+            for onset, duration, label in events:
+                writer.writeAnnotation(onset, duration, label)
+
         writer.setSignalHeaders(channel_info)
         writer.writeSamples(data_list)
 
+    except Exception as error:
+        logger.info(error)
+        raise error
+    finally:
+        writer.close()
+    return output_path
+
+
+def write_bdf(output_path: str,
+              raw_data: np.array,
+              ch_names: List[str],
+              sample_rate: float,
+              events: List[Tuple[float, float, str]],
+              overwrite: bool = False,
+              annotation_channels: int = 1) -> Path:
+    """
+    Converts BciPy raw_data to the BDF+ filetype using pyEDFlib.
+
+    Adapted from: https://github.com/holgern/pyedflib
+
+    Parameters
+    ----------
+    output_path - optional path to write converted data; defaults to writing
+        a file named raw.bdf in the raw_data_dir.
+    raw_data - raw data with a row for each channel
+    ch_names - names of the channels
+    sample_rate - sample frequency
+    events - List[Tuple(onset_in_seconds: float, duration_in_seconds: float, description: str)]
+    overwrite - If True, the destination file (if it exists) will be overwritten.
+        If False (default), an error will be raised if the file exists.
+    annotation_channels - number of annotation channels to use for writing triggers
+
+    Returns
+    -------
+        Path to new bdf file
+    """
+    if not overwrite and os.path.exists(output_path):
+        raise OSError('BDF file already exists.')
+
+    # set conversion parameters. The digital min/max are specific to BDF and much higher than EDF.
+    digital_min, digital_max = [-8388607, 8388607]
```

Comment:
The spec has the digital_min of -8388608.

However, edfwriter already has this information and initializes it in the `EdfWriter` constructor based on the provided `file_type` (see edfwriter.py line 209). We only need to provide specs for field values that we're overriding, so we actually don't need to include it here.